gradio/components.py

```diff
@@ -1230,28 +1231,115 @@ def __init__(
         """
         Parameters:
             choices: list of options to select from.
-            value: default value selected in dropdown. If None, no value is selected by default. If callable, the function will be called whenever the app loads to set the initial value of the component.
+            value: default value(s) selected in dropdown. If None, no value is selected by default. If callable, the function will be called whenever the app loads to set the initial value of the component.
             type: Type of value to be returned by component. "value" returns the string of the choice selected, "index" returns the index of the choice selected.
+            multiselect: if True, multiple choices can be selected. 
             label: component name in interface.
             every: If `value` is a callable, run the function 'every' number of seconds while the client connection is open. Has no effect otherwise. Queue must be enabled. The event can be accessed (e.g. to cancel it) via this component's .load_event attribute.
             show_label: if True, will display label.
             interactive: if True, choices in this dropdown will be selectable; if False, selection will be disabled. If not provided, this is inferred based on whether the component is used as an input or output.
             visible: If False, component will be hidden.
             elem_id: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
         """
-        Radio.__init__(
+        self.choices = choices or []
+        valid_types = ["value", "index"]
+        if type not in valid_types:
+            raise ValueError(
+                f"Invalid value for parameter `type`: {type}. Please choose from one of: {valid_types}"
+            )
+        self.type = type
+        self.multiselect = multiselect
+        self.test_input = self.choices[0] if len(self.choices) else None
+        self.interpret_by_tokens = False
+        IOComponent.__init__(
             self,
-            value=value,
-            choices=choices,
-            type=type,
             label=label,
             every=every,
             show_label=show_label,
             interactive=interactive,
             visible=visible,
             elem_id=elem_id,
+            value=value,
             **kwargs,
         )
+        self.cleared_value = self.value
+
+
+    def get_config(self):
+        return {
+            "choices": self.choices,
+            "value": self.value,
+            "multiselect": self.multiselect,
+            **IOComponent.get_config(self),
+        }
+
+    @staticmethod
+    def update(
+        value: Optional[Any] = _Keywords.NO_VALUE,
+        choices: Optional[str | List[str]] = None,
+        multiselect: bool = False,
```

Comment:
Better to make `multiselect=None` otherwise any update to a multi-select dropdown will switch off the multiselect

Reply:
We do not support `update` changing backend configuration yet, such as `multiselect`. `update` only allows changing frontend configuration. So please remove `multiselect` as an option to update